src/utils/src/FileIo.c

```diff
@@ -154,8 +153,8 @@ STATUS updateFile(PCHAR filePath, BOOL binMode, PBYTE pBuffer, UINT64 offset, UI
 
     CHK(0 == FSEEK(fp, (UINT32) offset, SEEK_SET), STATUS_INVALID_OPERATION);
 
-    for (i = 0, pCurPtr = pBuffer + offset; i < size; i++, pCurPtr++) {
-        CHK(EOF != FPUTC(*pCurPtr, fp), STATUS_WRITE_TO_FILE_FAILED);
+    for (i = 0; i < size; i++, pBuffer++) {
+        CHK(EOF != FPUTC(*pBuffer, fp), STATUS_WRITE_TO_FILE_FAILED);
```

Comment:
This is not the same as before right?  Before we start writing at `(pBuffer + offset)`, now we start writing at `pBuffer`.

Looks like it should be a while loop anyway but if we keep the for loop let's set the range of `i` to be from `offset` to `offset + size` and then in the FPUTC we can add `i` to pBuffer.  

A number of ways to do it, but this change is different than what was there before.